src/expression/definitions/format.ts

```diff
@@ -69,10 +70,21 @@ export class FormatExpression implements Expression {
                     if (!textColor) return null;
                 }
 
+                let verticalAlign = null;
+                if (arg['vertical-align']) {
+                    if (typeof arg['vertical-align'] === 'string' && !VERTICAL_ALIGN_OPTIONS.includes(arg['vertical-align'] as VerticalAlign)) {
```

Comment:
Can this if be moved to the parse code to keep this as clean as the other if statements?

Reply:
I did try to move it to parse - however then it shouldn't be placed in parse fn itself, but I found that it is better to add `enum` validation, like for array in other places.

However, to make it possible there are multiple changes that has to be made. Moreover the validation for enum is a bit different, because other validations are based on types, and enum validation requires access to value.

You can see how it looks here: https://github.com/tomtom-forks/maplibre-style-spec/pull/2

After making this try I actually think that having it in `format.ts`, as it is now, is better and much simpler. Also in similar fashion the validation is implemented for `interpolate` expression ([link to the file](https://github.com/maplibre/maplibre-style-spec/blob/93a8cd30b84b815ea57d49172959038a57a86759/src/expression/definitions/interpolate.ts#L71)).

Reply:
Yeah, the other PR is too complicated, and other places are doing the validation in the relevant class, much like resolvedImage is being done here.